tests/test_ufuncs.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import fast_numpy_loops
+import pytest
+fast_numpy_loops.initialize()
+
+
+rng = np.random.default_rng(123456)
+
+def type2dtype(types):
+    """
+    Maps the ufunc.type to the input, output dtypes.
+    type2dtype('ii->?') -> (np.int32, np.int32), (np.bool_,)
+    """
+    inp, out = types.split('->')
+    return tuple(np.dtype(c) for c in inp), tuple(np.dtype(c) for c in out)
+
+def get_ufuncs_and_types():
+    """Create a dictionary with keys of ufunc names and values of all the
+    supported type signatures
+    """
+    ufuncs = [x for x in dir(np) if isinstance(getattr(np, x), np.ufunc)]
+    # Maybe use a collections.defaultdict instead?
+    ret = dict([[x, []] for x in ufuncs])
+    for s in ret:
+        ret[s] = [type2dtype(t) for t in getattr(np, s).types]
+    return ret
```

Comment:
```suggestion
    return {s: [type2dtype(t) for t in getattr(np, s).types] for s in ufuncs}
```

more compact dictionary creation